middleware/validate-jwt.js

```diff
@@ -0,0 +1,26 @@
+const jwt = require('jsonwebtoken')
+const secret = require('../config').secret
+const User = require('../models/user')
+
+const validator = async function(req, res, next) {
+    const token = req.get('Authorization').split(' ')[1]
+    if(!token){
+        return res.status(401).json({message: 'no token provided'})
+    }
+    try{
+        const decoded = await jwt.verify(token, secret)
+        req.user = decoded
+        const user = await User.findById(decoded.user_id)
+        if(user == null){
```

Comment:
no need ti comapare null with ==. just put variable in if it will already consider false if it is null